src/azure-cli-core/azure/cli/core/test_utils/vcr_test_base.py

```diff
@@ -168,6 +171,30 @@ def _search_result_by_jmespath(json_data, query):
         json_val,
         jmespath.Options(collections.OrderedDict))
 
+def _custom_request_matcher(r1, r2):
+    """ Ensure method, path, and query parameters match. """
+    if r1.method != r2.method:
+        return False
+
+    url1 = urlparse(r1.uri)
+    url2 = urlparse(r2.uri)
+
+    if url1.path != url2.path:
+        return False
+
+    q1 = parse_qs(url1.query)
+    q2 = parse_qs(url2.query)
+    shared_keys = set(q1.keys()).union(set(q2.keys()))
```

Comment:
since this comes from `union` rather `intersection`, the name should be 'all_keys'